wordsearch/src/wordsearch.rs

```diff
@@ -91,21 +93,17 @@ impl WordSearch {
         let mut added_words = Vec::<Word>::new();
 
         while !words_by_len.is_empty() {
-            let current_word = words_by_len.get(0).unwrap().clone();
+            let current_word = words_by_len.get(0).unwrap();
 
-            if current_word.is_empty() {
-                break;
-            }
-
-            let mut possible_intersected_words = Vec::<Word>::with_capacity((size * size) as usize);
+            let mut possible_intersected_words = Vec::with_capacity((size * size) as usize);
 
             let errs = (&added_words).par_iter().map(|added_word| {
-                let mut collision_positions = Vec::<Point>::new();
+                let mut collision_positions: Vec<Point> = Vec::new();
```

Comment:
So in several places, I moved the type annotation to right after the variable, since it tends to make the complex part of the line a bit easier to read IMO. I think this is a preference I tend to see on most Rust codebases.